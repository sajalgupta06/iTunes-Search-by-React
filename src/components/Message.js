// @flow
import React, { PropTypes } from 'react';
import '../style/Message.css';

const msgMap: Object = {
  init: {
    icon: 'music_note',
    msg: 'Welcome back!'
  },
  loading: {
    msg: 'Loading...'
  },
  noContent: {
    icon: 'info',
    msg: 'No match'
  },
  error: {
    icon: 'error',
    msg: 'Error!'
  }
};

const spinner = _ => (
  <div className="sk-circle">
    <div className="sk-circle1 sk-child" />
    <div className="sk-circle2 sk-child" />
    <div className="sk-circle3 sk-child" />
    <div className="sk-circle4 sk-child" />
    <div className="sk-circle5 sk-child" />
    <div className="sk-circle6 sk-child" />
    <div className="sk-circle7 sk-child" />
    <div className="sk-circle8 sk-child" />
    <div className="sk-circle9 sk-child" />
    <div className="sk-circle10 sk-child" />
    <div className="sk-circle11 sk-child" />
    <div className="sk-circle12 sk-child" />
  </div>
);

const Message = ({ status }: { status: string }) => (
  <div className="toast">
    {status === 'loading' ? spinner() : <i className="material-icons">{msgMap[status].icon}</i>}
    <span className="text">{msgMap[status].msg}</span>
  </div>
);

export default Message;
