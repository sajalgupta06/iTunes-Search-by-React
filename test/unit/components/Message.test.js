import React from 'react';
import { shallow } from 'enzyme';
import Message from '../../../src/components/Message';

describe('<Message />', () => {
  test('Message component should renders correctly', () => {
    const props = {
      status: 'loading'
    };
    const component = shallow(<Message {...props} />);
    expect(component).toMatchSnapshot();
  });
});
