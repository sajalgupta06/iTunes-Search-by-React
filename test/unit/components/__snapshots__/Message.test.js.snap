// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Message component should renders correctly 1`] = `
<div
  className="toast"
>
  <div
    className="sk-circle"
  >
    <div
      className="sk-circle1 sk-child"
    />
    <div
      className="sk-circle2 sk-child"
    />
    <div
      className="sk-circle3 sk-child"
    />
    <div
      className="sk-circle4 sk-child"
    />
    <div
      className="sk-circle5 sk-child"
    />
    <div
      className="sk-circle6 sk-child"
    />
    <div
      className="sk-circle7 sk-child"
    />
    <div
      className="sk-circle8 sk-child"
    />
    <div
      className="sk-circle9 sk-child"
    />
    <div
      className="sk-circle10 sk-child"
    />
    <div
      className="sk-circle11 sk-child"
    />
    <div
      className="sk-circle12 sk-child"
    />
  </div>
  <span
    className="text"
  >
    Loading...
  </span>
</div>
`;
